python/cudf/cudf/tests/test_datetime.py

```diff
@@ -1300,6 +1300,38 @@ def test_is_leap_year():
     assert_eq(expect2, got2)
 
 
+def test_quarter():
+    data = [
+        "2020-05-31 08:00:00",
+        "1999-12-31 18:40:00",
+        "2000-12-31 04:00:00",
+        "1900-02-28 07:00:00",
+        "1800-03-14 07:30:00",
+        "2100-03-14 07:30:00",
+        "1970-01-01 00:00:00",
+        "1969-12-31 12:59:00",
+    ]
+    dtype = "datetime64[s]"
+
+    # Series
+    ps = pd.Series(data, dtype=dtype)
+    gs = cudf.from_pandas(ps)
+
+    expect = ps.dt.quarter
+    got = gs.dt.quarter.astype(np.int64)
+
+    assert_eq(expect, got)
```

Comment:
```suggestion
    got = gs.dt.quarter

    assert_eq(expect, got, check_dtype=False)
```

We could use `check_dtype=False` if we don't have to match pandas here.

Reply:
@beckernick would it be preferable here to mimic pandas completely, or to use something more memory efficient?

Reply:
We should save memory when we have guarantees around the maximum size, as we're often more memory constrained. Thanks for flagging this @galipremsagar @shaneding

Reply:
Sure. @shaneding, in that case can we leave libcudf API untouched and return `int8` types at python level.

Reply:
Shane shared that discussion with me as well. In this case, I think the performance penalty of typecasting for the `quarter` function is worth saving memory in Python workloads. 

We may want to think holistically about this discussion on Python side, if there hasn't already been a general pattern defined

Reply:
> We may want to think holistically about this discussion on Python side, if there hasn't already been a general pattern defined

We do this kind of memory saving on the python side for some normalization ops, replace ops, etc. with the help of following APIs:
`cudf.utils.dtypes.min_column_type`: https://github.com/rapidsai/cudf/blob/branch-21.10/python/cudf/cudf/utils/dtypes.py#L374
`cudf.utils.dtypes.min_float_dtype`: https://github.com/rapidsai/cudf/blob/branch-21.10/python/cudf/cudf/utils/dtypes.py#L404

We do this in some APIs already for `replace`, and all the APIs that use `_apply_support_method_axis_1` methods.

But in this specific case we could just hard-code `int8` as dtype.